ctapipe/core/selector.py

```diff
@@ -0,0 +1,126 @@
+"""
+Data Quality selection
+"""
+
+__all__ = ["Selector", "SelectionFunctionError"]
+
+from collections.abc import Callable
+
+import numpy as np
+
+from .component import Component
+from .traits import Dict
+
+
+class SelectionFunctionError(TypeError):
+    pass
+
+
+class Selector(Component):
+    """
+    Manages a set of user-configurable (at runtime or in a config file) selection
+    criteria that operate on the same type of input. Each time it is called, it
+    returns a boolean array of whether or not each criterion passed. It  also keeps
+    track of the total number of times each criterium is passed, as well as a
+    cumulative product of criterium (i.e. the criteria applied in-order)
+    """
+
+    selection_functions = Dict(
+        help=(
+            "dict of '<cut name>' : lambda function in string format to accept ("
+            "select) a given data value.  E.g. `{'mycut': 'lambda x: x > 3'}` "
+        )
+    ).tag(config=True)
+
+    def __init__(self, config=None, parent=None, **kwargs):
+        super().__init__(config=config, parent=parent, **kwargs)
+
+        # add a selection to count all entries and make it the first one
+        selection_functions = {"TOTAL": "lambda x: True"}
+        selection_functions.update(self.selection_functions)
+
+        self.selection_functions = selection_functions  # update
+
+        try:  # generate real functions from the selection function strings
+            self._selectors = {
+                name: eval(func_str) for name, func_str in selection_functions.items()
```

Comment:
add `eval(func_str, {}, {'np': np, 'u': u}) here, to make only numpy and astropy units available for func_str here.

Reply:
`import astropy.units as u` of course.

And a test showing that it fails with something like `subprocess.run(['echo', "about to erase 5 years of mc production"])` fails.